app/uk/gov/hmrc/excisemovementcontrolsystemapi/models/validation/MovementIdValidation.scala

```diff
@@ -0,0 +1,70 @@
+/*
+ * Copyright 2024 HM Revenue & Customs
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package uk.gov.hmrc.excisemovementcontrolsystemapi.models.validation
+
+import play.api.libs.json.Json
+import play.api.mvc.Result
+import play.api.mvc.Results.{BadRequest, NotFound}
+import uk.gov.hmrc.excisemovementcontrolsystemapi.models.ErrorResponse
+import uk.gov.hmrc.excisemovementcontrolsystemapi.repository.model.Movement
+import uk.gov.hmrc.excisemovementcontrolsystemapi.services.MovementService
+
+import java.time.Instant
+import java.util.UUID
+import javax.inject.Inject
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.Try
+
+case class MovementIdValidation @Inject()(movementService: MovementService)(implicit ec: ExecutionContext) {
+
+  def validateMovementId(id: String): Future[Either[MovementValidationError, Movement]] = {
```

Comment:
I would not call the movementService.getMovementId from here as this function does not just do the validation, but it retrieve the Id too. I would move the validation inside movementService.getMovementId or call the validator from movementService.getMovementId.